nixos/modules/services/networking/freeradius.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.freeradius;
+
+    freeradiusService = cfg: optionalAttrs cfg.enable {
+    freeradius = {
+      description = "FreeRadius server";
+      wantedBy = ["multi-user.target"];
+      after = ["network-online.target"];
+      requires = ["network-online.target"];
+      preStart = ''
+        ${cfg.configDir}/certs/bootstrap
+        ${pkgs.freeradius}/bin/radiusd -C -d ${cfg.configDir} -l stdout
+      '';
+
+      serviceConfig = {
+          ExecStart = "${pkgs.freeradius}/bin/radiusd -f -d ${cfg.configDir} -l stdout -xx";
+          ExecReload = [
+            "${pkgs.freeradius}/bin/radiusd -C -d ${cfg.configDir}  -l stdout"
+            "${pkgs.coreutils}/bin/kill -HUP $MAINPID"
+          ];
+          User = "radius";
+      };
+    };
+  };
+
+  freeradiusConfig = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable the freeradius server.
+      '';
+    };
+
+    configDir = mkOption {
+      type = types.nullOr types.path;
+      default = null;
```

Comment:
Why is this allowed to be null? This could default to `/var/lib/freeradius`.

Reply:
default value is set to /var/lib/freeradius. Without this parameter, the service will failed, so it makes sense to force default value. Default value for the `radiusd` command is `/etc/raddb`, should we use this ? I don't know about nixos policy on this.

Reply:
Probably yes. I believe freeradius never writes to this directory, does it? If it doesn't, `/etc` makes sense.
It's always good to find inspiration in other distros: https://git.archlinux.org/svntogit/community.git/tree/trunk/freeradius.service?h=packages/freeradius
Depending on whether freeradius needs write access to the filesystem, you should consider `ProtectSystem=strict` and `ReadWritePaths`: https://www.freedesktop.org/software/systemd/man/systemd.exec.html

Btw, please amend/squash the commit(s) instead of adding new ones.

Reply:
Done.

I used `ProtectSystem=full` because I had difficulties with `ReadWritePaths=` :
- the pre-start would fail `freeradius.service: Failed at step NAMESPACE spawning /nix/store/z0ppd98vmw50pc2fj6ycc71ayxvzyqsr-unit-script/bin/freeradius-pre-start: No such file or directory`
- I added ReadOnlyPaths="/nix/store" but it didn't solve the issue
- also, radius may need to write files (to /var/log/radius and maybe others), so it didn't seem wise (at least until we generate the freeradius configuration from nixos expressions).